File: Programowanie Obiektowe Projekt Nr 3 Python/Antelope.py
#-------------------------------------------------------------------------------
# Name:        module1
# Purpose:
#
# Author:      Kontekstowy
#
# Created:     27-05-2016
# Copyright:   (c) Kontekstowy 2016
# Licence:     <your licence>
#-------------------------------------------------------------------------------
from Animal import Animal
class Antelope(Animal):
    def __init__(self,xx,yy,world,numberx,tempSign):
        Animal.__init__(self,xx,yy,world,numberx,tempSign)
        self._initiative=4
        self._strength=4
        self._sign='A'

    def action(self,animalBoard,nx,ny,animalContainer,board):
        if animalBoard[self._x  + nx][self._y +ny] == -1:
            return 1
        if animalBoard[self._x + nx][self._y + ny] != -1:
            if animalBoard[self._x + nx][self._y + ny] == 'F':
                print("Antylopa zabija lisa")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'T':
                print("Zolw odpiera antylope")
                return 0
            if animalBoard[self._x + nx][self._y + ny] == 'A':
                if nx !=0 and ny !=0:
                    return 4
                else:
                    return 0
            if animalBoard[self._x + nx][self._y + ny] == 'W':
                print("Antylopa atakuje wilka lecz nie udaje sie to jej")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == 'H':
                print("Antylopa atakuje czlowieka lecz nie udaje sie to jej")
                return 3
            if animalBoard[self._x + nx][self._y + ny] == '#':
                print("Antylop zjada trawe")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'G':
                print("Antylopa zjada Guarane")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'N':
                print("Antylopa zjada trujace wilcze jagody i ginie")
                return 5
            if animalBoard[self._x + nx][self._y + ny] == 'L':
                print("Antylopa zjada mleczyk")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'S':
                print("Antylopa zjada owce")
                return 2

File: Programowanie Obiektowe Projekt Nr 2 Java/src/objectiveworld160553/Antelope.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objectiveworld160553;

/**
 *
 * @author Kontekstowy
 */
public class Antelope extends Animal {

    public Antelope(int x, int y, World world) {
        super(x, y, world);
        this.name = "A";
        this.initiative = 4;
        this.strength = 4;
    }

    @Override
    public int action(char animalBoard[][], int nx, int ny, Organism me) {
        switch (animalBoard[x + nx][y + ny]) {
            case '0':
                return 1;
            case 'A':
            //    System.out.println("Spotkaly sie 2 antylopy");
                return 4;
            case 'T':
                System.out.println("Zolw odpiera antylope");
                return 0;
            case 'N':
                System.out.println("Antylopa zjada wilcze jagody i ginie");
                return 5;
            case 'W':
                System.out.println("Wilk zabija antylope");
                return 3;
            case 'F':
                System.out.println("Antylopa zabija lisa");
                return 2;
            case 'L':
                System.out.println("Antylopa zjada mlecz");
                return 2;
            case '#':
                System.out.println("Antylopa zjada trawe");
                return 2;
            case 'G':
                System.out.println("Antylopa zjadla guaraney");
                return 2;
            case 'S':
                System.out.println("Spotyka sie antylopa z owca");
                return 0;
            case 'H':
                System.out.println("Antylopa atakuje czlowieka i ginie)");
                return 3;
        }

        return 0;
    }
}
